docs/adrs/0012-exposure-events-and-coenrolling-features.md

```diff
@@ -0,0 +1,142 @@
+# Exposure events and coenrolling features
+
+* Status: proposed
+* Deciders: dberry@mozilla.com, jhugman@mozilla.com, jlockhart@mozilla.com, shell@mozilla.com
+* Date: 2023-07-04, making the decision on or before 2023-07-11.
+
+Technical Story: [Supporting exposure events for coenrolled features](https://mozilla-hub.atlassian.net/browse/EXP-3602)
+
+## Context and Problem Statement
+
+Application feature code can be configured from and by Nimbus. In the case of `messaging` and `onboarding`, the complete set of messages and all their associated data is implemented by this feature configuration.
```

Comment:
May need to talk to Jared on https://mozilla-hub.atlassian.net/browse/EXP-3602 - which reads like co-enrollment should support other features. There’s too many options already - with sticky and first run and now multi-feature and rollouts. I don’t think we’re ready to expose this new highly complex pattern beyond messaging.

The change to allow co-enrollment should be the Messaging feature only. We should not change onboarding to support co-enrollment - or expose this to other features. This is a common pattern for messages only. It’s an exception and one I think we want to avoid for others in the foreseeable future. There is the one possible case for attribution based onboarding - but that is the exception and may be put on hold depending on [these discussions](https://docs.google.com/document/d/13ZFY7Dhprl5LE7tscdl1x9rAu8IPDJ3MN7mIQJraZBQ/edit#heading=h.5ipiksf3te2q).

Reply:
@jaredlockhart ^

Reply:
TBC: `allow-coenrollment` isn't covered in this ADR; however: as implemented in EXP-3602: it's in the feature manifest, and I expect will only be used under guidance.

Reply:
I'm on board so long as we are aligned that being in the feature manifest, but not exposing in Experimenter.  We need no-decision solution - set default on for Message Feature and default off for all others.

If it requires change - "using under guidance" has surpassed it's limits.  The one off support if we add another decision point and new scenario to other features is too expensive to make them ask me and then I ask eng :).  We need to learn the edges of co-enrollment with these 3 surfaces.  I cannot support onboarding even having it as an option until we have run coenrollment for a while and learned the use patterns and implications.

If we say "onboarding" in this document - it will become an assumed capability to use in the near term.

Reply:
>  we are aligned that being in the feature manifest

Yes.

> If we say "onboarding" in this document - it will become an assumed capability to use in the near term.

I have removed mention of `onboarding` in this document.